recipe/meta.yaml

```diff
@@ -14,7 +22,24 @@ source:
     - 3599.patch
 
 build:
-  number: 1
+  number: 2
+
+  # add build string so packages can depend on
+  # mpi or nompi variants
+  # dependencies:
+  # `pkg * mpi_mpich_*` for mpich
+  # `pkg * mpi_*` for any mpi
+  # `pkg * nompi_*` for no mpi
+  {% if mpi == 'nompi' %}
+  {% set mpi_prefix = "nompi" %}
+  {% else %}
+  {% set mpi_prefix = "mpi_" + mpi %}
+  {% endif %}
+  string: {{ mpi_prefix }}_h{{ PKG_HASH }}_{{ build }}
+
+  run_exports:
+    # strict runtime dependency on build-time MPI flavor
+    - {{ name }} * {{ mpi_prefix }}_*
```

Comment:
@minrk @isuruf AMReX uses a `YY.MM` version scheme and there are no ABI compatibility guarantees made. Should we add a specific run export here for this version scheme?

Reply:
That sounds like the right choice, yeah